server/src/main/java/org/elasticsearch/common/logging/ESLogMessage.java

```diff
@@ -72,4 +81,9 @@ public static String asJsonArray(Stream<String> stream) {
             .map(ESLogMessage::inQuotes)
             .collect(Collectors.joining(", ")) + "]";
     }
+
+    public static ESLogMessage message(String messagePattern, Object... args){
```

Comment:
Since we eventually collect the arguments using `.with(` we could probably structure this in a way that avoids repeating them. It's probably worth it even if we end up using a builder. 
I'm thinking something similar to: https://github.com/logstash/logstash-logback-encoder#event-specific-custom-fields